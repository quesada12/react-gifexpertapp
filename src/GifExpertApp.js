import React,{useState} from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = ( {defaultCategories = ['Shingeki No Kyojin']}) => {

    const [categories, setCategories] = useState(defaultCategories);

    // const handleAdd = () => {
    //     setCategories([...categories,'Dragon Ball Z']);
    //     setCategories( cat => [...cat,'Dragon Ball']);
    // }

    return (
        <>
          <h2>GifExpertApp</h2>
          <hr/>  

          <AddCategory setCategories= { setCategories } />

          <ol>
              { 
                categories.map( category => (
                    <GifGrid 
                        category={ category } 
                        key={ category } 
                    />
                )) 
              } 
          </ol>

        </>
    )
}

