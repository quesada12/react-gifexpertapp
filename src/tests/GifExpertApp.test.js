import { shallow } from "enzyme"
import { GifExpertApp } from "../GifExpertApp"

describe('Pruebas en el componente GifExpertApp', () => {
    
    test('Debe mostrarse correctamente', () => {
        
        const wrapper = shallow(<GifExpertApp />)
        expect(wrapper).toMatchSnapshot();

    })

    test('Debe mostrar una lista de categorias', () => {
        
        const categories = ['Naruto','Shikamaru'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>)
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })
    
    
})
