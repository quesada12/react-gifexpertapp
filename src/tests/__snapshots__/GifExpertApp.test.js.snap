// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifExpertApp Debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="Naruto"
      key="Naruto"
    />
    <GifGrid
      category="Shikamaru"
      key="Shikamaru"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en el componente GifExpertApp Debe mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="Shingeki No Kyojin"
      key="Shingeki No Kyojin"
    />
  </ol>
</Fragment>
`;
